From 9ba7459690674ce095847447db85acc1361c8153 Mon Sep 17 00:00:00 2001
From: Achim Thesmann <achim@google.com>
Date: Mon, 20 Oct 2025 14:15:14 -0700
Subject: [PATCH 06/38] Revert "Ignore pinned Windows"

Revert submission 32704936-b406880479

Reason for revert: app compat issue

Reverted changes: /q/submissionid:32704936-b406880479

Bug: 406880479
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:9aaf7b58c722cc019f55f49d3612062a499e73af
Merged-In: Iba3b8ab8e70200e0e8641a4e1210cf09c9205256
Change-Id: Iba3b8ab8e70200e0e8641a4e1210cf09c9205256
---
 .../wm/BackgroundActivityStartController.java |  23 +---
 .../wm/BackgroundLaunchProcessController.java |  11 +-
 .../wm/VisibleActivityProcessTracker.java     |  12 +-
 .../server/wm/WindowProcessController.java    |   3 +-
 ...ActivityStartControllerExemptionTests.java | 114 ++----------------
 ...ackgroundActivityStartControllerTests.java |   4 -
 ...ackgroundLaunchProcessControllerTests.java |  38 ++----
 .../VisibleActivityProcessTrackerTests.java   |  99 ---------------
 8 files changed, 26 insertions(+), 278 deletions(-)
 delete mode 100644 services/tests/wmtests/src/com/android/server/wm/VisibleActivityProcessTrackerTests.java

diff --git a/services/core/java/com/android/server/wm/BackgroundActivityStartController.java b/services/core/java/com/android/server/wm/BackgroundActivityStartController.java
index 2d2c8c477249..f8a50b3fda04 100644
--- a/services/core/java/com/android/server/wm/BackgroundActivityStartController.java
+++ b/services/core/java/com/android/server/wm/BackgroundActivityStartController.java
@@ -309,7 +309,6 @@ public class BackgroundActivityStartController {
         private final int mCallingPid;
         private final @ActivityTaskManagerService.AppSwitchState int mAppSwitchState;
         private final boolean mCallingUidHasVisibleActivity;
-        private final boolean mCallingUidHasVisibleNotPinnedActivity;
         private final boolean mCallingUidHasNonAppVisibleWindow;
         private final @ActivityManager.ProcessState int mCallingUidProcState;
         private final boolean mIsCallingUidPersistentSystemProcess;
@@ -319,7 +318,6 @@ public class BackgroundActivityStartController {
         private final int mRealCallingUid;
         private final int mRealCallingPid;
         private final boolean mRealCallingUidHasVisibleActivity;
-        private final boolean mRealCallingUidHasVisibleNotPinnedActivity;
         private final boolean mRealCallingUidHasNonAppVisibleWindow;
         private final @ActivityManager.ProcessState int mRealCallingUidProcState;
         private final boolean mIsRealCallingUidPersistentSystemProcess;
@@ -410,8 +408,6 @@ public class BackgroundActivityStartController {
                     mCallingUidProcState <= ActivityManager.PROCESS_STATE_PERSISTENT_UI;
             mCallingUidHasVisibleActivity =
                     getService().mVisibleActivityProcessTracker.hasVisibleActivity(callingUid);
-            mCallingUidHasVisibleNotPinnedActivity = getService().mVisibleActivityProcessTracker
-                    .hasVisibleNotPinnedActivity(callingUid);
             mCallingUidHasNonAppVisibleWindow = getService().mActiveUids.hasNonAppVisibleWindow(
                     callingUid);
             if (realCallingUid == NO_PROCESS_UID) {
@@ -419,13 +415,11 @@ public class BackgroundActivityStartController {
                 mRealCallingUidProcState = PROCESS_STATE_NONEXISTENT;
                 mRealCallingUidHasVisibleActivity = false;
                 mRealCallingUidHasNonAppVisibleWindow = false;
-                mRealCallingUidHasVisibleNotPinnedActivity = false;
                 mRealCallerApp = null;
                 mIsRealCallingUidPersistentSystemProcess = false;
             } else if (callingUid == realCallingUid) {
                 mRealCallingUidProcState = mCallingUidProcState;
                 mRealCallingUidHasVisibleActivity = mCallingUidHasVisibleActivity;
-                mRealCallingUidHasVisibleNotPinnedActivity = mCallingUidHasVisibleNotPinnedActivity;
                 mRealCallingUidHasNonAppVisibleWindow = mCallingUidHasNonAppVisibleWindow;
                 // In the PendingIntent case callerApp is not passed in, so resolve it ourselves.
                 mRealCallerApp = callerApp == null
@@ -437,9 +431,6 @@ public class BackgroundActivityStartController {
                 mRealCallingUidHasVisibleActivity =
                         getService().mVisibleActivityProcessTracker.hasVisibleActivity(
                                 realCallingUid);
-                mRealCallingUidHasVisibleNotPinnedActivity =
-                        getService().mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(
-                                realCallingUid);
                 mRealCallingUidHasNonAppVisibleWindow =
                         getService().mActiveUids.hasNonAppVisibleWindow(realCallingUid);
                 mRealCallerApp = getService().getProcessController(realCallingPid, realCallingUid);
@@ -573,8 +564,6 @@ public class BackgroundActivityStartController {
             sb.append("; callingPid: ").append(mCallingPid);
             sb.append("; appSwitchState: ").append(mAppSwitchState);
             sb.append("; callingUidHasVisibleActivity: ").append(mCallingUidHasVisibleActivity);
-            sb.append("; callingUidHasVisibleNotPinnedActivity: ")
-                    .append(mCallingUidHasVisibleNotPinnedActivity);
             sb.append("; callingUidHasNonAppVisibleWindow: ").append(
                     mCallingUidHasNonAppVisibleWindow);
             sb.append("; callingUidProcState: ").append(DebugUtils.valueToString(
@@ -609,8 +598,6 @@ public class BackgroundActivityStartController {
                 sb.append("; realCallingPid: ").append(mRealCallingPid);
                 sb.append("; realCallingUidHasVisibleActivity: ")
                         .append(mRealCallingUidHasVisibleActivity);
-                sb.append("; realCallingUidHasVisibleNotPinnedActivity: ")
-                        .append(mRealCallingUidHasVisibleNotPinnedActivity);
                 sb.append("; realCallingUidHasNonAppVisibleWindow: ")
                         .append(mRealCallingUidHasNonAppVisibleWindow);
                 sb.append("; realCallingUidProcState: ").append(DebugUtils.valueToString(
@@ -1050,9 +1037,8 @@ public class BackgroundActivityStartController {
         final boolean appSwitchAllowedOrFg = state.mAppSwitchState == APP_SWITCH_ALLOW
                 || state.mAppSwitchState == APP_SWITCH_FG_ONLY
                 || isHomeApp(state.mCallingUid, state.mCallingPackage);
-        if (appSwitchAllowedOrFg && state.mCallingUidHasVisibleNotPinnedActivity) {
-            return new BalVerdict(BAL_ALLOW_VISIBLE_WINDOW,
-                    "callingUid has visible non-pinned window");
+        if (appSwitchAllowedOrFg && state.mCallingUidHasVisibleActivity) {
+            return new BalVerdict(BAL_ALLOW_VISIBLE_WINDOW, "callingUid has visible window");
         }
         return BalVerdict.BLOCK;
     };
@@ -1180,9 +1166,8 @@ public class BackgroundActivityStartController {
         final boolean appSwitchAllowedOrFg = state.mAppSwitchState == APP_SWITCH_ALLOW
                 || state.mAppSwitchState == APP_SWITCH_FG_ONLY
                 || isHomeApp(state.mRealCallingUid, state.mRealCallingPackage);
-        if (appSwitchAllowedOrFg && state.mRealCallingUidHasVisibleNotPinnedActivity) {
-            return new BalVerdict(BAL_ALLOW_VISIBLE_WINDOW,
-                    "realCallingUid has visible non-pinned window");
+        if (appSwitchAllowedOrFg && state.mRealCallingUidHasVisibleActivity) {
+            return new BalVerdict(BAL_ALLOW_VISIBLE_WINDOW, "realCallingUid has visible window");
         }
         return BalVerdict.BLOCK;
     };
diff --git a/services/core/java/com/android/server/wm/BackgroundLaunchProcessController.java b/services/core/java/com/android/server/wm/BackgroundLaunchProcessController.java
index d9d638604df3..2605310db6f4 100644
--- a/services/core/java/com/android/server/wm/BackgroundLaunchProcessController.java
+++ b/services/core/java/com/android/server/wm/BackgroundLaunchProcessController.java
@@ -120,8 +120,7 @@ class BackgroundLaunchProcessController {
     BalVerdict areBackgroundActivityStartsAllowed(
             int pid, int uid, String packageName,
             int appSwitchState, BalCheckConfiguration checkConfiguration,
-            boolean hasActivityInVisibleTask, boolean inPinnedWindow,
-            boolean hasBackgroundActivityStartPrivileges,
+            boolean hasActivityInVisibleTask, boolean hasBackgroundActivityStartPrivileges,
             long lastStopAppSwitchesTime, long lastActivityLaunchTime,
             long lastActivityFinishTime) {
         // Allow if the proc is instrumenting with background activity starts privs.
@@ -148,11 +147,9 @@ class BackgroundLaunchProcessController {
                     : BAL_ALLOW_VISIBLE_WINDOW, /*background*/
                     "process bound by foreground uid");
         }
-        // Allow if the caller has an activity in any foreground task, unless it's a pinned window
-        // and not a foreground service start.
-        if ((checkConfiguration.isCheckingForFgsStart || !inPinnedWindow)
-                && checkConfiguration.checkOtherExemptions
-                && hasActivityInVisibleTask && appSwitchState != APP_SWITCH_DISALLOW) {
+        // Allow if the caller has an activity in any foreground task.
+        if (checkConfiguration.checkOtherExemptions && hasActivityInVisibleTask
+                && appSwitchState != APP_SWITCH_DISALLOW) {
             return new BalVerdict(BAL_ALLOW_FOREGROUND, /*background*/
                     "process has activity in foreground task");
         }
diff --git a/services/core/java/com/android/server/wm/VisibleActivityProcessTracker.java b/services/core/java/com/android/server/wm/VisibleActivityProcessTracker.java
index 6b39239a1be5..0e2b71c02637 100644
--- a/services/core/java/com/android/server/wm/VisibleActivityProcessTracker.java
+++ b/services/core/java/com/android/server/wm/VisibleActivityProcessTracker.java
@@ -80,22 +80,14 @@ class VisibleActivityProcessTracker {
      * {@link ActivityRecord#mVisibleRequested} or {@link ActivityRecord#isVisible()} is true.
      */
     boolean hasVisibleActivity(int uid) {
-        return match(uid, WindowContainer.alwaysTruePredicate());
-    }
-
-    /**
-     * Returns {@code true} if the uid has a process that contains an activity with
-     * {@link ActivityRecord#mVisibleRequested} or {@link ActivityRecord#isVisible()} is true.
-     */
-    boolean hasVisibleNotPinnedActivity(int uid) {
-        return match(uid, wpc -> !wpc.inPinnedWindowingMode());
+        return match(uid, null /* predicate */);
     }
 
     private boolean match(int uid, Predicate<WindowProcessController> predicate) {
         synchronized (mProcMap) {
             for (int i = mProcMap.size() - 1; i >= 0; i--) {
                 final WindowProcessController wpc = mProcMap.keyAt(i);
-                if (wpc.mUid == uid && predicate.test(wpc)) {
+                if (wpc.mUid == uid && (predicate == null || predicate.test(wpc))) {
                     return true;
                 }
             }
diff --git a/services/core/java/com/android/server/wm/WindowProcessController.java b/services/core/java/com/android/server/wm/WindowProcessController.java
index 8fdd135296e5..d356128205df 100644
--- a/services/core/java/com/android/server/wm/WindowProcessController.java
+++ b/services/core/java/com/android/server/wm/WindowProcessController.java
@@ -727,8 +727,7 @@ public class WindowProcessController extends ConfigurationContainer<Configuratio
             int appSwitchState, BalCheckConfiguration checkConfiguration) {
         return mBgLaunchController.areBackgroundActivityStartsAllowed(mPid, mUid,
                 mInfo.packageName, appSwitchState, checkConfiguration,
-                hasActivityInVisibleTask(), inPinnedWindowingMode(),
-                mInstrumentingWithBackgroundActivityStartPrivileges,
+                hasActivityInVisibleTask(), mInstrumentingWithBackgroundActivityStartPrivileges,
                 mAtm.getLastStopAppSwitchesTime(),
                 mLastActivityLaunchTime, mLastActivityFinishTime);
     }
diff --git a/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerExemptionTests.java b/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerExemptionTests.java
index 70337610f11c..e3a8776aca6a 100644
--- a/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerExemptionTests.java
+++ b/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerExemptionTests.java
@@ -50,7 +50,6 @@ import android.platform.test.annotations.Presubmit;
 import android.platform.test.annotations.RequiresFlagsEnabled;
 import android.platform.test.flag.junit.CheckFlagsRule;
 import android.platform.test.flag.junit.DeviceFlagsValueProvider;
-import android.platform.test.flag.junit.SetFlagsRule;
 import android.provider.DeviceConfig;
 import android.util.Pair;
 import android.view.WindowManager;
@@ -65,7 +64,6 @@ import com.android.window.flags.Flags;
 
 import org.junit.After;
 import org.junit.Before;
-import org.junit.ClassRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
@@ -90,10 +88,6 @@ import java.util.Set;
 @RunWith(JUnit4.class)
 public class BackgroundActivityStartControllerExemptionTests {
 
-    @ClassRule
-    public static final SetFlagsRule.ClassRule mClassRule = new SetFlagsRule.ClassRule();
-    @Rule public final SetFlagsRule mSetFlagsRule = mClassRule.createSetFlagsRule();
-
     private static final int REGULAR_UID_1 = 10100;
     private static final int REGULAR_UID_2 = 10200;
     private static final int NO_UID = -1;
@@ -271,8 +265,6 @@ public class BackgroundActivityStartControllerExemptionTests {
 
         // setup state
         when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(callingUid))).thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(callingUid)))
-                .thenReturn(false);
         when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
 
         // prepare call
@@ -285,7 +277,6 @@ public class BackgroundActivityStartControllerExemptionTests {
                 originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
                 checkedOptions);
         assertThat(balState.toString()).contains("callingUidHasVisibleActivity: true");
-        assertThat(balState.toString()).contains("callingUidHasVisibleNotPinnedActivity: false");
         assertThat(balState.toString()).contains("callingUidHasNonAppVisibleWindow: false");
 
         // call
@@ -294,47 +285,8 @@ public class BackgroundActivityStartControllerExemptionTests {
         balState.setResultForCaller(callerVerdict);
 
         // assertions
-        assertWithMessage(balState + " -> " + callerVerdict)
-                .that(callerVerdict.getCode())
-                .isEqualTo(BAL_BLOCK);
-    }
-
-    @Test
-    public void testCaller_appHasVisiblePinnedWindow() {
-        int callingUid = REGULAR_UID_1;
-        int callingPid = REGULAR_PID_1;
-        final String callingPackage = REGULAR_PACKAGE_1;
-        int realCallingUid = REGULAR_UID_2;
-        int realCallingPid = REGULAR_PID_2;
-
-        // setup state
-        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(callingUid))).thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(callingUid)))
-                .thenReturn(true);
-        when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
-
-        // prepare call
-        PendingIntentRecord originatingPendingIntent = mPendingIntentRecord;
-        boolean allowBalExemptionForSystemProcess = false;
-        Intent intent = TEST_INTENT;
-        ActivityOptions checkedOptions = mCheckedOptions;
-        BackgroundActivityStartController.BalState balState = mController.new BalState(callingUid,
-                callingPid, callingPackage, realCallingUid, realCallingPid, mCallerApp,
-                originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
-                checkedOptions);
-        assertThat(balState.toString()).contains("callingUidHasVisibleActivity: true");
-        assertThat(balState.toString()).contains("callingUidHasVisibleNotPinnedActivity: true");
-        assertThat(balState.toString()).contains("callingUidHasNonAppVisibleWindow: false");
-
-        // call
-        BalVerdict callerVerdict = mController.checkBackgroundActivityStartAllowedByCaller(
-                balState);
-        balState.setResultForCaller(callerVerdict);
-
-        // assertions
-        assertWithMessage(balState + " -> " + callerVerdict)
-                .that(callerVerdict.getCode())
-                .isEqualTo(BAL_ALLOW_VISIBLE_WINDOW);
+        assertWithMessage(balState.toString()).that(callerVerdict.getCode()).isEqualTo(
+                BAL_ALLOW_VISIBLE_WINDOW);
     }
 
     @Test
@@ -346,10 +298,8 @@ public class BackgroundActivityStartControllerExemptionTests {
         int realCallingPid = REGULAR_PID_2;
 
         // setup state
-        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(realCallingUid)))
-                .thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(realCallingUid)))
-                .thenReturn(false);
+        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(realCallingUid))).thenReturn(
+                true);
         when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
 
         // prepare call
@@ -362,8 +312,6 @@ public class BackgroundActivityStartControllerExemptionTests {
                 originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
                 checkedOptions);
         assertThat(balState.toString()).contains("realCallingUidHasVisibleActivity: true");
-        assertThat(balState.toString())
-                .contains("realCallingUidHasVisibleNotPinnedActivity: false");
         assertThat(balState.toString()).contains("realCallingUidHasNonAppVisibleWindow: false");
 
         // call
@@ -372,48 +320,8 @@ public class BackgroundActivityStartControllerExemptionTests {
         balState.setResultForRealCaller(realCallerVerdict);
 
         // assertions
-        assertWithMessage(balState + " -> " + realCallerVerdict)
-                .that(realCallerVerdict.getCode())
-                .isEqualTo(BAL_BLOCK);
-    }
-
-    @Test
-    public void testRealCaller_appHasNotPinnedVisibleWindow() {
-        int callingUid = REGULAR_UID_1;
-        int callingPid = REGULAR_PID_1;
-        final String callingPackage = REGULAR_PACKAGE_1;
-        int realCallingUid = REGULAR_UID_2;
-        int realCallingPid = REGULAR_PID_2;
-
-        // setup state
-        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(realCallingUid)))
-                .thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(realCallingUid)))
-                .thenReturn(true);
-        when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
-
-        // prepare call
-        PendingIntentRecord originatingPendingIntent = mPendingIntentRecord;
-        boolean allowBalExemptionForSystemProcess = false;
-        Intent intent = TEST_INTENT;
-        ActivityOptions checkedOptions = mCheckedOptions;
-        BackgroundActivityStartController.BalState balState = mController.new BalState(callingUid,
-                callingPid, callingPackage, realCallingUid, realCallingPid, mCallerApp,
-                originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
-                checkedOptions);
-        assertThat(balState.toString()).contains("realCallingUidHasVisibleActivity: true");
-        assertThat(balState.toString()).contains("realCallingUidHasVisibleNotPinnedActivity: true");
-        assertThat(balState.toString()).contains("realCallingUidHasNonAppVisibleWindow: false");
-
-        // call
-        BalVerdict realCallerVerdict = mController.checkBackgroundActivityStartAllowedByRealCaller(
-                balState);
-        balState.setResultForRealCaller(realCallerVerdict);
-
-        // assertions
-        assertWithMessage(balState + " -> " + realCallerVerdict)
-                .that(realCallerVerdict.getCode())
-                .isEqualTo(BAL_ALLOW_VISIBLE_WINDOW);
+        assertWithMessage(balState.toString()).that(realCallerVerdict.getCode()).isEqualTo(
+                BAL_ALLOW_VISIBLE_WINDOW);
     }
 
     @Test
@@ -497,8 +405,6 @@ public class BackgroundActivityStartControllerExemptionTests {
 
         // setup state
         when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(callingUid))).thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(callingUid)))
-                .thenReturn(true);
         when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
 
         // prepare call
@@ -513,7 +419,6 @@ public class BackgroundActivityStartControllerExemptionTests {
                 originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
                 checkedOptions);
         assertThat(balState.toString()).contains("callingUidHasVisibleActivity: true");
-        assertThat(balState.toString()).contains("callingUidHasVisibleNotPinnedActivity: true");
         assertThat(balState.toString()).contains("callingUidHasNonAppVisibleWindow: false");
 
         // call
@@ -536,10 +441,8 @@ public class BackgroundActivityStartControllerExemptionTests {
         int realCallingPid = REGULAR_PID_2;
 
         // setup state
-        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(realCallingUid)))
-                .thenReturn(true);
-        when(mVisibleActivityProcessTracker.hasVisibleNotPinnedActivity(eq(realCallingUid)))
-                .thenReturn(true);
+        when(mVisibleActivityProcessTracker.hasVisibleActivity(eq(realCallingUid))).thenReturn(
+                true);
         when(mService.getBalAppSwitchesState()).thenReturn(APP_SWITCH_ALLOW);
 
         // prepare call
@@ -554,7 +457,6 @@ public class BackgroundActivityStartControllerExemptionTests {
                 originatingPendingIntent, allowBalExemptionForSystemProcess, mResultRecord, intent,
                 checkedOptions);
         assertThat(balState.toString()).contains("realCallingUidHasVisibleActivity: true");
-        assertThat(balState.toString()).contains("realCallingUidHasVisibleNotPinnedActivity: true");
         assertThat(balState.toString()).contains("realCallingUidHasNonAppVisibleWindow: false");
 
         // call
diff --git a/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerTests.java b/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerTests.java
index 3abe0b8a6fa8..fe9a6e746513 100644
--- a/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerTests.java
+++ b/services/tests/wmtests/src/com/android/server/wm/BackgroundActivityStartControllerTests.java
@@ -563,7 +563,6 @@ public class BackgroundActivityStartControllerTests {
                         + "callingPid: 11001; "
                         + "appSwitchState: 0; "
                         + "callingUidHasVisibleActivity: false; "
-                        + "callingUidHasVisibleNotPinnedActivity: false; "
                         + "callingUidHasNonAppVisibleWindow: false; "
                         + "callingUidProcState: NONEXISTENT; "
                         + "isCallingUidPersistentSystemProcess: false; "
@@ -582,7 +581,6 @@ public class BackgroundActivityStartControllerTests {
                         + "realCallingUid: 1; "
                         + "realCallingPid: 1; "
                         + "realCallingUidHasVisibleActivity: false; "
-                        + "realCallingUidHasVisibleNotPinnedActivity: false; "
                         + "realCallingUidHasNonAppVisibleWindow: false; "
                         + "realCallingUidProcState: NONEXISTENT; "
                         + "isRealCallingUidPersistentSystemProcess: false; "
@@ -667,7 +665,6 @@ public class BackgroundActivityStartControllerTests {
                         + "callingPid: 11001; "
                         + "appSwitchState: 0; "
                         + "callingUidHasVisibleActivity: false; "
-                        + "callingUidHasVisibleNotPinnedActivity: false; "
                         + "callingUidHasNonAppVisibleWindow: false; "
                         + "callingUidProcState: NONEXISTENT; "
                         + "isCallingUidPersistentSystemProcess: false; "
@@ -686,7 +683,6 @@ public class BackgroundActivityStartControllerTests {
                         + "realCallingUid: 1; "
                         + "realCallingPid: 1; "
                         + "realCallingUidHasVisibleActivity: false; "
-                        + "realCallingUidHasVisibleNotPinnedActivity: false; "
                         + "realCallingUidHasNonAppVisibleWindow: false; "
                         + "realCallingUidProcState: NONEXISTENT; "
                         + "isRealCallingUidPersistentSystemProcess: false; "
diff --git a/services/tests/wmtests/src/com/android/server/wm/BackgroundLaunchProcessControllerTests.java b/services/tests/wmtests/src/com/android/server/wm/BackgroundLaunchProcessControllerTests.java
index a439178600ba..615370d6bb9c 100644
--- a/services/tests/wmtests/src/com/android/server/wm/BackgroundLaunchProcessControllerTests.java
+++ b/services/tests/wmtests/src/com/android/server/wm/BackgroundLaunchProcessControllerTests.java
@@ -103,7 +103,6 @@ public class BackgroundLaunchProcessControllerTests {
                     /* checkOtherExemptions */ true,
                     ACTIVITY_BG_START_GRACE_PERIOD_MS);
     boolean mHasActivityInVisibleTask = false;
-    boolean mInPinnedWindoMode = false;
     boolean mHasBackgroundActivityStartPrivileges = false;
     long mLastStopAppSwitchesTime = 0L;
     long mLastActivityLaunchTime = 0L;
@@ -114,8 +113,7 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_BLOCK);
@@ -127,8 +125,7 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_PERMISSION);
@@ -164,8 +161,7 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_TOKEN);
@@ -197,8 +193,7 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_TOKEN);
@@ -228,38 +223,20 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_BOUND_BY_FOREGROUND);
     }
 
-    @Test
-    public void testForegroundTaskBlockedIfPinned() {
-        mAppSwitchState = APP_SWITCH_ALLOW;
-        mHasActivityInVisibleTask = true;
-        mInPinnedWindoMode = true;
-        BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
-                mPid, mUid, mPackageName,
-                mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
-                mLastStopAppSwitchesTime, mLastActivityLaunchTime,
-                mLastActivityFinishTime);
-        assertThat(balVerdict.getCode()).isEqualTo(BAL_BLOCK);
-    }
-
     @Test
     public void testForegroundTask() {
         mAppSwitchState = APP_SWITCH_ALLOW;
         mHasActivityInVisibleTask = true;
-        mInPinnedWindoMode = false;
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_FOREGROUND);
@@ -275,8 +252,7 @@ public class BackgroundLaunchProcessControllerTests {
         BalVerdict balVerdict = mController.areBackgroundActivityStartsAllowed(
                 mPid, mUid, mPackageName,
                 mAppSwitchState, mBalCheckConfiguration,
-                mHasActivityInVisibleTask, mInPinnedWindoMode,
-                mHasBackgroundActivityStartPrivileges,
+                mHasActivityInVisibleTask, mHasBackgroundActivityStartPrivileges,
                 mLastStopAppSwitchesTime, mLastActivityLaunchTime,
                 mLastActivityFinishTime);
         assertThat(balVerdict.getCode()).isEqualTo(BAL_ALLOW_GRACE_PERIOD);
diff --git a/services/tests/wmtests/src/com/android/server/wm/VisibleActivityProcessTrackerTests.java b/services/tests/wmtests/src/com/android/server/wm/VisibleActivityProcessTrackerTests.java
deleted file mode 100644
index 7e2d6fb74f5d..000000000000
--- a/services/tests/wmtests/src/com/android/server/wm/VisibleActivityProcessTrackerTests.java
+++ /dev/null
@@ -1,99 +0,0 @@
-/*
- * Copyright (C) 2022 The Android Open Source Project
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.android.server.wm;
-
-import static android.content.pm.ActivityInfo.INSETS_DECOUPLED_CONFIGURATION_ENFORCED;
-
-import static com.google.common.truth.Truth.assertThat;
-
-import static org.mockito.Mockito.doReturn;
-import static org.mockito.Mockito.mock;
-
-import android.app.IApplicationThread;
-import android.app.WindowConfiguration;
-import android.content.pm.ApplicationInfo;
-import android.platform.test.annotations.Presubmit;
-
-import androidx.test.filters.SmallTest;
-
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-
-
-/**
- * Tests for the {@link com.android.server.wm.VisibleActivityProcessTracker} class.
- *
- * Build/Install/Run:
- * atest WmTests:VisibleActivityProcessTrackerTests
- */
-@SmallTest
-@Presubmit
-@RunWith(WindowTestRunner.class)
-public class VisibleActivityProcessTrackerTests extends WindowTestsBase {
-
-    private VisibleActivityProcessTracker mTracker;
-
-    @Before
-    public void setup() {
-        mTracker = mAtm.mVisibleActivityProcessTracker;
-    }
-
-    @Test
-    public void testVisibleActivity() {
-        WindowProcessController wpc = createWindowProcessController();
-        assertThat(mTracker.hasVisibleActivity(wpc.mUid)).isFalse();
-        mTracker.onAnyActivityVisible(wpc);
-        assertThat(mTracker.hasVisibleActivity(wpc.mUid)).isTrue();
-        mTracker.onAllActivitiesInvisible(wpc);
-        assertThat(mTracker.hasVisibleActivity(wpc.mUid)).isFalse();
-    }
-
-    @Test
-    public void testVisibleNotPinnedActivity() {
-        WindowProcessController wpc = createWindowProcessController();
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isFalse();
-        mTracker.onAnyActivityVisible(wpc);
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isTrue();
-        mTracker.onAllActivitiesInvisible(wpc);
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isFalse();
-    }
-
-    @Test
-    public void testVisiblePinnedActivity() {
-        WindowProcessController wpc = createWindowProcessController();
-        wpc.getConfiguration().windowConfiguration.setWindowingMode(
-                WindowConfiguration.WINDOWING_MODE_PINNED);
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isFalse();
-        mTracker.onAnyActivityVisible(wpc);
-        assertThat(mTracker.hasVisibleActivity(wpc.mUid)).isTrue();
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isFalse();
-        mTracker.onAllActivitiesInvisible(wpc);
-        assertThat(mTracker.hasVisibleNotPinnedActivity(wpc.mUid)).isFalse();
-    }
-
-    WindowProcessController createWindowProcessController() {
-        WindowProcessListener mMockListener = mock(WindowProcessListener.class);
-        ApplicationInfo info = mock(ApplicationInfo.class);
-        info.packageName = "test.package.name";
-        doReturn(true).when(info).isChangeEnabled(INSETS_DECOUPLED_CONFIGURATION_ENFORCED);
-        WindowProcessController mWpc = new WindowProcessController(
-                mAtm, info, null, 0, -1, null, mMockListener);
-        mWpc.setThread(mock(IApplicationThread.class));
-        return mWpc;
-    }
-}
-- 
2.53.0

